src/components/TypingTest/TyingBox.tsx

```diff
@@ -0,0 +1,73 @@
+import { Dispatch, SetStateAction, useMemo } from "react";
+import Character from "./Character";
+import { KBLayoutType } from "../../enums/KBLayout";
+
+type Props = {
+  sentence: string;
+  userInput: string;
+  setUserInput: Dispatch<SetStateAction<string>>;
+  sourceLayout: KBLayoutType;
+  targetLayout: KBLayoutType;
+};
+
+function TypingBox(props: Props) {
+  const { sentence, userInput, setUserInput, sourceLayout, targetLayout } =
+    props;
+
+  function handleKeyDown(event: { key: string }) {
+    if (event.key == "Backspace") {
+      setUserInput((userInput) => {
+        return userInput.slice(0, -1);
+      });
+    } else if (event.key == "Shift") {
+    } else {
+      setUserInput((userInput) => {
+        return userInput + event.key;
+      });
+    }
+  }
+
+  function convertString(
+    input: string,
+    source: KBLayoutType,
+    target: KBLayoutType
+  ) {
+    if (source == null || target == null) return input;
+
+    const mapping: { [key: string]: string } = {};
+    for (let i = 0; i < source.length; i++) {
+      mapping[source[i]] = target[i];
+    }
+    return input
```

Comment:
I prefer keeping this as a map and using that to get the mappedChar. That way, we don't actually need to convert the whole sentence